### PR TITLE
fix: parse operand value when field is number type

### PR DESCRIPTION
**Pull request by Jshhhh**
## Problem
comparing equality with filter operand as string returns false. 0 == '0'

## Solution
Parse operand value when field type is a number

### Ticket

GitHub issue _____

### Automated tests
- [x] Unit tests added/updated
- [ ] E2E tests added/updated
- [ ] N/A - (provide a reason)
- [ ] deferred - (provide GitHub issue for tracking)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react.test.ts.snap [hunk 1]
```diff
@@ -1094,7 +1094,7 @@ export default function CollectionOfCustomButtons(
   } = props;
   const itemsFilterObj = {
     and: [
-      { field: \\"age\\", operand: \\"10\\", operator: \\"gt\\" },
+      { field: \\"age\\", operand: 10, operator: \\"eq\\" },
```
Comment (cwoolum) [inline packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react.test.ts.snap:1097]: The test changed. Was this intentional?

**Review by cwoolum — APPROVED** [review]

File: packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react.test.ts.snap (hunk 1 above)
Comment (Jshhhh) [reply packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react.test.ts.snap:1097]: Yes. Expecting the operand to be number and not a string

File: packages/codegen-ui-react/lib/__tests__/react-component-render-helper.test.ts [hunk 2]
```diff
@@ -345,4 +346,15 @@ describe('react-component-render-helper', () => {
       assertASTMatchesSnapshot(exp);
     });
   });
+
+  describe('parseNumberOperand', () => {
+    test('should parse int if field data type is Int', () => {
+      expect(parseNumberOperand('10', { dataType: 'Int', readOnly: false, required: false, isArray: false })).toBe(10);
+    });
+    test('should parse int if field data type is Int', () => {
```
Comment (rtpascual) [inline packages/codegen-ui-react/lib/__tests__/react-component-render-helper.test.ts:354]: nit: the test name should change to `Float`

Comment (Jshhhh) [reply packages/codegen-ui-react/lib/__tests__/react-component-render-helper.test.ts:354]: ah damn haha. ill fix it in another pr

**Pull request merged**